### PR TITLE
Coverity nitpicks

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

File: tests/valgrind/potential-memleaks.c [hunk 1]
```diff
@@ -100,7 +100,8 @@ pl_workthread_strerror_localtime(void * cookie)
 	(void)cookie; /* UNUSED */
 	(void)str; /* UNUSED */
 
-	time(&now);
+	if (time(&now))
```
Comment (cperciva) [inline tests/valgrind/potential-memleaks.c:103]: What's the return value of `time(3)`?

Comment (gperciva) [reply tests/valgrind/potential-memleaks.c:103]: "any of the reasons described in gettimeofday(2)".

... and yes, I should fix the commit message.

Comment (gperciva) [reply tests/valgrind/potential-memleaks.c:103]: ... ah, `man 3 time` does go on to specify that neither C99 nor POSIX.1 requires time() to see errno on failure.

Comment (cperciva) [reply tests/valgrind/potential-memleaks.c:103]: That too, but I said return value, not errno.

**Comment by gperciva** [comment]
Rebased to check `time()` as done in `aws/aws_sign.c`.  Also fixed commit subject line.

**Pull request merged**